File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Compass Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var c = new Tinkerforge.BrickletCompass(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Set period for heading callback to 0.1s (100ms) without a threshold
        c.setHeadingCallbackConfiguration(100, false, 'x', 0, 0);
    }
);

// Register heading callback
c.on(Tinkerforge.BrickletCompass.CALLBACK_HEADING,
    // Callback function for heading callback
    function (heading) {
        console.log('Heading: ' + heading/10.0 + ' °');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/ruby/example_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_compass'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Compass Bricklet

ipcon = IPConnection.new # Create IP connection
c = BrickletCompass.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register heading callback
c.register_callback(BrickletCompass::CALLBACK_HEADING) do |heading|
  puts "Heading: #{heading/10.0} °"
end

# Set period for heading callback to 0.1s (100ms) without a threshold
c.set_heading_callback_configuration 100, false, 'x', 0, 0

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
